mmaction/models/recognizers/recognizer2d.py

```diff
@@ -70,20 +72,31 @@ def _do_test(self, imgs):
             x = x.squeeze(2)
             num_segs = 1
 
+        if self.feature_extraction:
+            # perform spatial pooling
+            avg_pool = nn.AdaptiveAvgPool2d(1)
```

Comment:
_do_test will be called multi times in recognize2d & 3d, maybe we can move avg_pool to __init__

Reply:
How about `F.adaptive_avg_pool2d`